libraries/tock-register-interface/src/macros.rs

```diff
@@ -115,7 +116,7 @@ macro_rules! register_bitmasks {
                     match v {
                         $(
                             $(#[$inner])*
-                            x if x == Value::$valname as $valtype => Some(Value::$valname),
+                            _x if _x == $value => Some(Value::$valname),
```

Comment:
The `repr($valtype)` is necessary to avoid warnings when storing a value greater than `isize::MAX`, because the default enum representation is `isize`. see: https://github.com/tock/tock/pull/1823 . 

I think it is also necessary to store `u64` without values being truncated, though I can't remember if I tested that.

Reply:
Oh, I see..